base/docs/Docs.jl

```diff
@@ -373,28 +373,31 @@ function summarize(io::IO, λ::Function, binding)
     println(io, "```\n", methods(λ), "\n```")
 end
 
-function summarize(io::IO, U::UnionAll, binding)
-    summarize(io, Base.unwrap_unionall(U), binding)
-    println(io, "Note, ```", Base.typename(U),
-            "``` is represented by a wrapping type ```UnionAll``` that " *
-            "describes the bounds of its type parameters.\n")
+function summarize(io::IO, T::Type{Union}, binding)
+    println(io, "**Types:**")
+    t = getfield(binding.mod, binding.var)
+    for ut in Core.Inference.uniontypes(Base.unwrap_unionall(t))
+        print(io, " - `")
+        Base.show_datatype(io, isa(ut, UnionAll) ? Base.unwrap_unionall(ut) : ut)
```

Comment:
I don't really understand the purpose of this code. We already have `show` methods for types; why not just call `show`?